doc/internal/man3/OSSL_TIME.pod

```diff
@@ -0,0 +1,90 @@
+=pod
+
+=head1 NAME
+
+OSSL_TIME, OSSL_TIME_SECOND, OSSL_TIME_NEVER,
+ossl_time_now, ossl_time_time_to_timeval, ossl_time_compare,
+ossl_time_add, ossl_time_subtract
+- times and durations
+
+=head1 SYNOPSIS
+
+ #include "internal/time.h"
+
+ typedef OSSL_TIME;
+
+ #define OSSL_TIME_SECOND
+ #define OSSL_TIME_NEVER
+
+ OSSL_TIME ossl_time_now(void);
+ void ossl_time_time_to_timeval(OSSL_TIME t, struct timeval *out);
+
+ int ossl_time_compare(OSSL_TIME a, OSSL_TIME b);
+ OSSL_TIME ossl_time_add(OSSL_TIME a, OSSL_TIME b);
+ OSSL_TIME ossl_time_subtract(OSSL_TIME a, OSSL_TIME b);
+
+=head1 DESCRIPTION
+
+These functions allow the current time to be obtained and for basic
+arithmetic operations to be safely performed with times and durations.
+
+B<OSSL_TIME> is a type definition that holds a time.  It should not be
+assumed to be a structure, union or basic type.
+
+B<OSSL_TIME_SECOND> is an integer that indicates the precision of an
+B<OSSL_TIME>.  Specifically, it is the number of counts per second that
+a time can represent.
+
+B<OSSL_TIME_NEVER> is the largest representable B<OSSL_TIME>.  This value
+is returned when an overflow would otherwise occur.
+
+B<ossl_time_now> returns the current time.
+
+B<ossl_time_time_to_timeval> converts a time to a I<struct timeval>.
+
+B<ossl_time_compare> compares I<a> with I<b> and returns -1 if I<a>
+is smaller than I<b>, 0 if they are equal and +1 if I<a> is
+larger than I<b>.
+
+B<ossl_time_add> performs a saturating addition of the two times,
+returning I<a> + I<b>.
+If the summation would overflow B<OSSL_TIME_NEVER> is returned.
+
+B<ossl_time_subtract> performs a saturating subtraction of the two items,
+returning I<a> - I<b>.
+If the difference would be negative, the I<Epoch> (i.e. B<0>) is returned.
+
+=head1 NOTES
+
+The I<Epoch> is the start of the year 1970 UTC.  Times before this date
+are not supported.
+
```

Comment:
I think there's some confusion here about whether OSSL_TIME represents a duration or a point in time.

For example, you have a function to convert to a timeval. timeval can represent a duration or a point in time. `gettimeofday` outputs a point in time timeval, but e.g. `select` takes a timeval representing a duration, the timeout.
 Like `struct timeval`, `struct timespec` can also represent a duration (`ppoll` timeout) or a point in time (`clock_gettime`). In the latter case though there are multiple possible clocks (epochs) so the `struct timespec` only has semantic meaning in the context of some contextually known epoch, rather than a specific fixed epoch.

I believe OSSL_TIME should be usable for representing both points in time and durations as this is the usual convention with POSIX.

I would probably put something like "OSSL_TIME can represent a duration, or a point in time. Where it is used to represent a point in time, it does so by expressing a duration relative to some reference epoch. The OSSL_TIME structure itself does not contain information about the epoch used; thus, interpretation of an OSSL_TIME requires that the epoch it is to be interpreted relative to is contextually understood."